Streamline insight save

mbwatson opened the pull request:
This PR isn't tied to an issue, just conversation around tightening up the save experience.

This involves relocating the Save button alongside other actions in the toolbar and moving the form into a modal.

<img width="1668" height="1115" alt="image" src="https://github.com/user-attachments/assets/28819ca6-2b0d-4adb-bbaf-5e4eaade014c" />

Additionally, a new Summary markdown field is added.

edit: this experience has shape-shifted a bit after a few discussions and iterations. see latest comments for up-to-date screenshots

mbwatson committed:
move save form to modal

style existing inputs

mbwatson committed:
add summary field to insight saved fields

support md in pdf summary

remove debug logging

mbwatson committed:
render md with narkdown.markdown

mbwatson committed:
support md editor min_height prop

mbwatson committed:
emphasize primary action button

mbwatson committed:
remove obsolete save form component

davidborland reviewed (commented):
I really like the markdown preview. I'm less sure about the modal-based workflow though. I think it is nice to be able to see and edit the current versions before saving. I.e. I might adjust some things in the vis, write some text, decide I want to add an annotation, go back to the text, etc. I'm curious what @davegotz thinks, as he has just gone through the process of creating some insights.

mbwatson commented:
ah, yes -- good point. i will rethink!

edit: it might be worth noting that you can cancel and come back to the modal without losing changes

davegotz commented:
> I really like the markdown preview. I'm less sure about the modal-based workflow though. I think it is nice to be able to see and edit the current versions before saving. I.e. I might adjust some things in the vis, write some text, decide I want to add an annotation, go back to the text, etc. I'm curious what @davegotz thinks, as he has just gone through the process of creating some insights.

Agreed. It's important to be able to go back and forth with the visualization.

mbwatson committed:
Merge branch 'dev' into enhancement/insight-save

mbwatson committed:
drawerify save form, styles

mbwatson commented:
i reworked the save insight form into a lower drawer. it's not complete yet, but this already feels better, as it still allows access to the underlying content while the save form is present. feel free to weight in on this direction, @davidborland, @davegotz

davidborland commented:
I like this better than the modal. Two things I think could be improved:
1. I don't like that you have to click the Save button to bring up the text content to edit. I think this should be a separate control.
2. It is unclear if clicking the X in the drawer cancels or keeps changes. 

I think a separate control to bring up the drawer would probably solve 1 and 2, and then the Save button could maybe just bring up a simple confirm/cancel dialog.

What do you think about that?

mbwatson commented:
i think this makes sense. i had been considering the text entries as the final step before saving, almost like a "Save as.."-then-choose-a-filename kind of pattern. however, it sounds like you are imagining the text entries as just another control(s), alongside the existing ones. does that sound right? i think i like that better, so i will keep tinkering along those lines.

davidborland commented:
Yeah, I can imagine there being some back and forth between editing the description and the visualization before finalizing it.

mbwatson committed:
fit entire image in insight card

mbwatson committed:
Merge branch 'dev' into enhancement/insight-save

mbwatson committed:
add contrast bwetween card and app backgrounds

throw in select borders tweak

mbwatson committed:
create collapsible card component

mbwatson committed:
use collapsible cards in controls ui

mbwatson committed:
setup modal-based save experience

mbwatson commented:
here's the current state. as always, feedback is welcome

[Screencast from 2025-12-11 12-16-21.webm](https://github.com/user-attachments/assets/24a33d08-4272-4214-8143-1edb5602da32)

quick thoughts about some nice-to-haves:
- summaries of the info in the card when in the collapsed state
- drag-to-reorder

mbwatson committed:
pull in initial summary text

mbwatson committed:
remove old save component

mbwatson committed:
emphasize primary action

mbwatson committed:
improve card look, usability

mbwatson committed:
remove redundant md editor labels

mbwatson committed:
align md editor styles with dmc input styles

mbwatson committed:
add placeholders, improve controls styles

add 'no annotations' text placeholder

improve style

add placeholder to title field

mbwatson committed:
re-implement annotations hiding in boxplot explorer

mbwatson committed:
soften app bg/card contrast

mbwatson committed:
fit entire image in insight card

mbwatson committed:
resize insight card thumbnails

align insight and custom insight card components

mbwatson committed:
Merge branch 'dev' into 107-thumbnail-fit

mbwatson committed:
Merge branch 'dev' into enhancement/insight-save

mbwatson committed:
tweak card contents breakpoint

mbwatson committed:
wip

mbwatson committed:
bring in summary, add style

mbwatson committed:
align title input style

mbwatson committed:
separate chart controls, metadata inputs into tabs

mbwatson committed:
move callbacks to separate file; add icons, style

mbwatson commented:
latest change on this branch: insight form fields (chart controls and metadata inputs) into separated into two tabs.

here's a quick recording to show the new pattern in action:

[Screencast from 2025-12-15 09-37-52.webm](https://github.com/user-attachments/assets/0ec035ba-ab14-4c3e-87e0-34b355003309)

mbwatson committed:
remove additional lower padding

mbwatson committed:
put layout in separate card

mbwatson committed:
remove obsolete file

mbwatson committed:
move css into cs dir

mbwatson committed:
import dash.exceptions

mbwatson committed:
add fallback in case of missing summary field

mbwatson committed:
fix custom insight detail stacking

mbwatson committed:
Merge branch '107-thumbnail-fit' into enhancement/insight-save

mbwatson committed:
relocate save button, rename tabs, add intro save note

mbwatson committed:
separate card actions from overall click event

mbwatson committed:
improve save button style

mbwatson commented:
chart controls tab
<img width="1548" height="1171" alt="Screenshot from 2025-12-17 10-28-26" src="https://github.com/user-attachments/assets/863c0cb5-2be2-4f47-8eae-643167c3d67c" />

save insight tab
<img width="1548" height="1171" alt="Screenshot from 2025-12-17 10-28-29" src="https://github.com/user-attachments/assets/ec87dbc5-024d-42d1-b485-6a14176b1b92" />

confirmation after initial save click
<img width="1548" height="1171" alt="Screenshot from 2025-12-17 10-28-32" src="https://github.com/user-attachments/assets/ba0fd45f-042d-4661-b3da-ac43dd5ed9a7" />

davidborland reviewed (changes requested):
I like it. The only thing I might change is the wording of this:

"Additional information is required to save this custom insight. Please complete the fields below to proceed."

Maybe just 

"Edit the fields below to update the information for this insight"

?

mbwatson committed:
create insight save+confirm button

mbwatson committed:
add interval timout in save flow

delete suplicate css file

mbwatson committed:
Merge branch 'dev' into enhancement/insight-save

mbwatson committed:
remove unused component

mbwatson committed:
simplify save form guidance
